File: programs/ifactor_1000000000000037/Go-trial-odd-int_Go/ifactor.go
// Copyright (C) 2016, Florent Gallaire <fgallaire@gmail.com>
// MIT license

package main

import "os"
import "fmt"
import "strings"
import "strconv"

func main() {
    N := os.Args[1]
    n, _ := strconv.Atoi(N)
    dk := 2
    p := []string{}
    for n > 1 {
        q := n / dk
        r := n % dk
        if r == 0 {
            p = append(p, strconv.Itoa(dk))
            n = q
        } else if q > dk {
            if dk == 2 {
                dk = 3
            } else {
                dk = dk + 2
            }
        } else {
            p = append(p, strconv.Itoa(n))
            break
        }        
    }
    fmt.Println(N + ": " + strings.Join(p, " "))
}

File: programs/ifactor_1000000000000037/Julia-trial-odd_Julia/ifactor.jl
# Copyright (C) 2017, Florent Gallaire <fgallaire@gmail.com>
# MIT license

N = ARGS[1]
n = parse(BigInt, N)
dk = 2
p = []

while n > 1
    q, r = divrem(n, dk)
    if r == 0
        push!(p, dk)
        n = q
    elseif q > dk
        if dk == 2
            dk = 3
        else
            dk = dk + 2
        end
    else
        push!(p, n)
        break
    end
end

println(N * ": " * join(p, " "))
